File: src/cpp/twosum.cpp
#include <iostream>
#include <string>
#include <vector>
#include <list>
#include <map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    	map<int, int> m;
	vector<int> ret;
	for (int i=0; i<nums.size(); i++){
	    if (m.find(nums[i])== m.end()  )
		m[target-nums[i]] = i;
	    else {
		ret.push_back(m[nums[i]] + 1);
		ret.push_back(i+1);
		break;
	    }
	}  
	return ret;  
    }
};

int main(){
    vector<int> v ;
    v.push_back(2);
    v.push_back(7);
    v.push_back(11);
    v.push_back(15);
    int target = 9;
    Solution s ;
    vector<int> vv = s.twoSum(v, target);
    for (vector<int>::iterator it = v.begin(); it!=v.end(); it++)
	cout<<*it<<endl;    
    for (auto n:vv)
	cout<<n<<endl;
}

File: src/python/twosum.py

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
	m = {}
	ret = None
        for i,j in enumerate(nums):
		if j not in m:
			m[target-j] = i
		else:
			ret = (m[j]+1, i+1)
	return [i for i in ret]

if __name__ == '__main__':
  	s = Solution()
	nums = [2,7,11,15]
	target = 9
	print [i for i in s.twoSum(nums, target)]	
